connectors/email/src/main/java/io/camunda/connector/email/core/jakarta/JakartaActionExecutor.java

```diff
@@ -0,0 +1,342 @@
+            .limit(imapListEmails.getMaxToBeRead())
+            .toList();
+      }
+    } catch (MessagingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private DeleteEmailResponse pop3DeleteEmail(
+      Pop3DeleteEmail pop3DeleteEmail, Authentication authentication, Session session) {
+    try (Store store = session.getStore()) {
+      this.jakartaUtils.connectStore(store, authentication);
+      try (Folder folder = store.getFolder("INBOX")) {
+        return deleteEmail(folder, pop3DeleteEmail.getMessageId());
+      }
+    } catch (MessagingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ReadEmailResponse pop3ReadEmail(
+      Pop3ReadEmail pop3ReadEmail, Authentication authentication, Session session) {
+    try {
+      try (Store store = session.getStore()) {
+        this.jakartaUtils.connectStore(store, authentication);
+        try (Folder folder = store.getFolder("INBOX")) {
+          folder.open(Folder.READ_WRITE);
+          Message[] messages = folder.search(new MessageIDTerm(pop3ReadEmail.getMessageId()));
+          return Arrays.stream(messages)
+              .findFirst()
+              .map(Email::createEmail)
+              .map(
+                  email ->
+                      new ReadEmailResponse(
+                          email.getMessageId(),
+                          email.getFrom(),
+                          email.getSubject(),
+                          email.getSize(),
+                          email.getBody().getBodyAsPlainText(),
+                          email.getBody().getBodyAsHtml()))
+              .orElseThrow(() -> new MessagingException("No emails have been found with this ID"));
+        }
+      }
+    } catch (MessagingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private List<ListEmailsResponse> pop3ListEmails(
+      Pop3ListEmails pop3ListEmails, Authentication authentication, Session session) {
+    try {
+      try (Store store = session.getStore()) {
+        this.jakartaUtils.connectStore(store, authentication);
+        try (Folder folder = store.getFolder("INBOX")) {
+          folder.open(Folder.READ_ONLY);
+          return Arrays.stream(folder.getMessages())
+              .map(Email::createBodylessEmail)
+              .sorted(
+                  this.jakartaUtils.retrieveEmailComparator(
+                      pop3ListEmails.getSortField(), pop3ListEmails.getSortOrder()))
+              .map(
+                  email ->
+                      new ListEmailsResponse(
+                          email.getMessageId(),
+                          email.getFrom(),
+                          email.getSubject(),
+                          email.getSize()))
+              .limit(pop3ListEmails.getMaxToBeRead())
+              .toList();
+        }
+      }
+    } catch (MessagingException e) {
+      throw new RuntimeException(e);
```

Comment:
consider adding some error logs (if we can show some valuable information, if not ignore my comment).
Same for other exceptions thrown